### PR TITLE
verification working

### PR DESCRIPTION
**Pull request by BlessingAmoah**
The verification of email which is my technical challenge works

### CODE REVIEW

File: Frontend/student-hub/node_modules/.cache/.eslintcache
Comment (Mustafa-Sadriwala) [inline Frontend/student-hub/node_modules/.cache/.eslintcache:1]: if you have `.gitignore` file you should add the Frontend/student-hub/node_modules/.cache to it as these files do not need to be reviewed/added to the repo

File: backend/routes/profile.js [hunk 1]
```diff
@@ -20,16 +20,16 @@ router.get('/', async (req, res) => {
   }
 });
 
-router.post('/', verifyToken, async (req, res) => {
+router.put('/', verifyToken, async (req, res) => {
   const {
-    name, profilePicture, bio, phone, address, city, state, school, course, interest
+    name, profilePicture, bio,email, phone, address, city, state, school, major, interest, mentorship
```
Comment (tpv0613) [inline backend/routes/profile.js:25]: If we're allowing users to update their email, would we re-send a verification email to the new email?

**Comment by tpv0613** [comment]
Love the progress you've been making! 

But I noticed that the pull request is quite large with a lot of code changes unrelated to email verification, such as implementing the frontend and backend for some of the profile and dashboard. This makes it harder to review the code relevant to email verification, which from your title and comment seems to be the focus of this pull request.

In the future, I would recommend making the pull requests more focused on smaller chunks of code that can be tested without needing other changes :)

File: Frontend/student-hub/src/components/Navbar.js [hunk 2]
```diff
@@ -85,17 +85,14 @@ function Navbar({ isLoggedIn, setIsLoggedIn}) {
...
-                                </MenuItem>
                                 <MenuItem onClick={() => { navigate('/feedback'); handleSettingsClose(); }}>Feedback</MenuItem>
                                 <MenuItem onClick={() => { navigate('/contact'); handleSettingsClose(); }}>Contact Me</MenuItem>
                                 <MenuItem onClick={() => { navigate('/about'); handleSettingsClose(); }}>About</MenuItem>
                             </Menu>
                     </>
                 ) : (
                     <>
+
                     </>
```
Comment (Mustafa-Sadriwala) [inline Frontend/student-hub/src/components/Navbar.js:96]: instead of an empty <></> you should use `undefined`

File: Frontend/student-hub/src/pages/Dashboard.js [hunk 3]
```diff
@@ -33,6 +33,11 @@ function Dashboard() {
     const [searchTerm, setSearchTerm] = useState('');
     const [filteredData, setFilteredData] = useState(null);
     const [openCreateModal, setOpenCreateModal] = useState(false);
+    const [openCommentsModal, setOpenCommentsModal] = useState(false);
+    const [currentComments, setCurrentComments] = useState([]);
+    const [currentPostId, setCurrentPostId] = useState(null);
+    const [newComment, setNewComment] = useState('');
+
```
Comment (Mustafa-Sadriwala) [inline Frontend/student-hub/src/pages/Dashboard.js:40]: since this isn't relevant to Verification this could be in a separate PR/Commit

File: Frontend/student-hub/src/pages/CoursePage.js
Comment (Mustafa-Sadriwala) [inline Frontend/student-hub/src/pages/CoursePage.js:1]: changes here aren't relevant to verification so should be in a distinct PR

**Review by Mustafa-Sadriwala — APPROVED** [review]
Overall the code makes sense, however there are many changes here unrelated to verification.

Going to approve this since the code is working, but also think we should follow-up on Thanh's point about whether emails can be changed (I think this will need to be an un-editable field)

File: backend/routes/profile.js (hunk 1 above)
Comment (BlessingAmoah) [reply backend/routes/profile.js]: I have made changes to this so the user can not edit that part. Thanks for pointing me to this.

File: Frontend/student-hub/node_modules/.cache/.eslintcache
Comment (BlessingAmoah) [reply Frontend/student-hub/node_modules/.cache/.eslintcache:1]: I have created a .gitignore file and placed the node_modules in but it still shows after the commit. So I tried looking at options which one was to use git rm --cached node_modules but still not working.

File: Frontend/student-hub/src/components/Navbar.js (hunk 2 above)
Comment (BlessingAmoah) [reply Frontend/student-hub/src/components/Navbar.js:96]: I have made this change

File: Frontend/student-hub/src/pages/Dashboard.js (hunk 3 above)
Comment (BlessingAmoah) [reply Frontend/student-hub/src/pages/Dashboard.js:40]: Ok I will do that.
